src/server/server.ts

```diff
@@ -132,6 +133,7 @@ type ToolHandler<
   extra: RequestHandlerExtra<ServerRequest, ServerNotification>,
 ) => TReturn | Promise<TReturn>;
 
+// biome-ignore lint/complexity/noBannedTypes: Required for proper type inference and literal type preservation
```

Comment:
You can use `Record<never, ToolDef>` as a default type. This satisfies the constraint while ensuring no key exists by default